src/downloader.rs

```diff
@@ -36,34 +37,78 @@ impl Downloader {
             .build();
 
         // Download the files asynchronously.
-        let _tasks = stream::iter(downloads)
+        stream::iter(downloads)
             .map(|d| self.fetch(&client, d))
             .buffer_unordered(self.concurrent_downloads)
             .collect::<Vec<_>>()
-            .await;
+            .await
     }
 
     /// Fetches the files and write them to disk.
-    async fn fetch(&self, client: &ClientWithMiddleware, download: &Download) -> Result<(), Error> {
+    async fn fetch(&self, client: &ClientWithMiddleware, download: &Download) -> Summary {
+        // Create a download summary.
+        let mut summary = Summary::new(download.clone(), StatusCode::BAD_REQUEST, 0);
+
         // Request the file.
         debug!("Fetching {}", download.url.clone());
-        let res = client.get(download.url.clone()).send().await.unwrap();
+        let res = match client.get(download.url.clone()).send().await {
+            Ok(res) => res,
+            Err(e) => {
+                return summary.with_status(Status::Fail(e.to_string()));
```

Comment:
is summary in this case an `Result::Ok` variant?

Reply:
No, not in this case. If the GET operation succeeds, it means let's use the response and try to download the bytes. If it fails, we want to return the summary and the failure reason (invalid DNS, timeout, etc.).